apps/antalmanac/src/components/AppBar/TimeFormatMenu.tsx

```diff
@@ -0,0 +1,83 @@
+import { withStyles } from '@material-ui/core/styles';
+import { ClassNameMap } from '@material-ui/core/styles/withStyles';
+import AccessTimeIcon from '@material-ui/icons/AccessTime';
+import { PureComponent } from 'react';
+
+import { toggleShow24HourTime } from '$actions/AppStoreActions';
+import AppStore from '$stores/AppStore';
+
+const styles = {
+    container: {
+        padding: '0.5rem',
+        minWidth: '12.25rem',
+    },
+    betaBadge: { transform: 'scale(1) translate(95%, -50%)' },
+};
+
+interface TimeFormatSettingsState {
+    anchorEl?: HTMLElement;
+    show24HourTime: boolean;
+}
+
+class TimeFormatMenu extends PureComponent<{ classes: ClassNameMap }, TimeFormatSettingsState> {
+    state: TimeFormatSettingsState = {
+        anchorEl: undefined,
+        show24HourTime: AppStore.getShow24HourTime(),
+    };
+
+    componentDidMount = () => {
+        AppStore.on('show24HourToggle', () => {
+            this.setState({ show24HourTime: AppStore.getShow24HourTime() });
+        });
+    };
+
+    render() {
+        const { classes } = this.props;
+
+        return (
+            <>
+                <Button
+                    onClick={(event) => {
+                        this.setState({ anchorEl: event.currentTarget });
+                    }}
+                    color="inherit"
+                    startIcon={<AccessTimeIcon />}
+                >
+                    Time Format
+                </Button>
+                <Popover
+                    open={Boolean(this.state.anchorEl)}
+                    anchorEl={this.state.anchorEl}
+                    onClose={() => {
+                        this.setState({ anchorEl: undefined });
+                    }}
+                    anchorOrigin={{
+                        vertical: 'bottom',
+                        horizontal: 'center',
+                    }}
+                    transformOrigin={{
+                        vertical: 'top',
+                        horizontal: 'center',
+                    }}
+                >
+                    <Paper className={classes.container}>
+                        <FormControl>
+                            <RadioGroup
+                                aria-label="show24HourTime"
+                                name="show24HourTime"
+                                value={this.state.show24HourTime.toString()}
+                                onChange={toggleShow24HourTime}
+                            >
+                                <FormControlLabel value="false" control={<Radio color="primary" />} label="12-hour" />
+                                <FormControlLabel value="true" control={<Radio color="primary" />} label="24-hour" />
+                            </RadioGroup>
+                        </FormControl>
+                    </Paper>
+                </Popover>
+            </>
+        );
+    }
+}
+
+export default withStyles(styles)(TimeFormatMenu);
```

Comment:
Since this is a new component, try to use write it with a functional component and using MUIv5.
Also, I prefer named exports over default exports because it's easier for code autocompletion to find those names and autocomplete the imports.

```suggestion
import { useCallback, useState } from 'react';
import { Button, FormControl, FormControlLabel, Paper, Popover, Radio, RadioGroup } from '@mui/material';
import { AccessTime as AccessTimeIcon } from '@mui/icons-material';
import { useTimeFormatStore } from '$stores/time-format'

export function TimeFormatMenu() {
    const [anchorEl, setAnchorEl] = useState<HTMLElement>();

    const [timeFormat, setTimeFormat] = useTimeFormatStore(store => [store.timeFormat, store.setTimeFormat])

    const handleClick = useCallback((event: React.MouseEvent<HTMLButtonElement>) => {
        setAnchorEl(event.currentTarget);
    }, []);

    const handleClose = useCallback(() => {
        setAnchorEl(undefined);
    }, []);

    const handleChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
        setTimeFormat(event.target.value === 'true')
    }, [])

    return (
        <>
            <Button color="inherit" onClick={handleClick} startIcon={<AccessTimeIcon />}>
                Time Format
            </Button>
            <Popover
                open={Boolean(anchorEl)}
                anchorEl={anchorEl}
                onClose={handleClose}
                anchorOrigin={{
                    vertical: 'bottom',
                    horizontal: 'center',
                }}
                transformOrigin={{
                    vertical: 'top',
                    horizontal: 'center',
                }}
            >
                <Paper sx={{ padding: 0.5, minWidth: 12 }}>
                    <FormControl>
                        <RadioGroup
                            aria-label="show24HourTime"
                            name="show24HourTime"
                            value={timeFormat}
                            onChange={handleChange}
                        >
                            <FormControlLabel value="false" control={<Radio color="primary" />} label="12-hour" />
                            <FormControlLabel value="true" control={<Radio color="primary" />} label="24-hour" />
                        </RadioGroup>
                    </FormControl>
                </Paper>
            </Popover>
        </>
    );
}
```